### ISSUE TITLE
Handle simd intrinsics

### ISSUE THREAD

**Issue by jrmuizel** [immunant/c2rust#28]
For example it would be nice to be able to handle the following code:

```c
#include <xmmintrin.h>
#include <emmintrin.h>

void
unpack_128_2x128 (__m128i data, __m128i* data_lo, __m128i* data_hi)
{
    *data_lo = _mm_unpacklo_epi8 (data, _mm_setzero_si128 ());
    *data_hi = _mm_unpackhi_epi8 (data, _mm_setzero_si128 ());
}
```

**Comment by rinon** [comment]
Indeed. We decided a while back to put simd support off until someone needed it. Considering simd support is in stable now, we'll take another look at translating simd types and builtins. Assuming everything matches up well enough, we should be able to add this soon.

**Comment by ahomescu** [comment]
Translating the types and builtins directly to their Rust [`core::arch` equivalents](https://doc.rust-lang.org/1.27.0/core/arch/x86_64/index.html) looks like the most straightforward approach.

**Comment by TheDan64** [comment]
SIMD support has been merged as of 78372fd634d1a8227a5caf03ea8fa46e79a5e639. The translator should now translate all of the `__m*` types and most of the `__mm*` functions (rust is missing some small [subset of instructions](https://github.com/rust-lang-nursery/stdsimd/issues/579) so we can't translate those)

If you find any other SIMD functions that go untranslated, and have rust equivalents (even if not stable), please feel free to reopen this issue!
